### ISSUE TITLE
 cannot find symbol  startForeground(FOREGROUND_SERVICE_TYPE_MICROPHONE, notification);

### ISSUE THREAD

**Issue by ajitdas123** [react-native-webrtc/react-native-callkeep#327]
# Bug report

- [x] I've checked the [example](https://github.com/react-native-webrtc/react-native-callkeep/tree/master/example) to reproduce the issue.

At the time building the app, the package throwing error saying 

```
/Users/ad/MY COOL APP/node_modules/react-native-callkeep/android/src/main/java/io/wazo/callkeep/VoiceConnectionService.java:59: error: cannot find symbol
import static android.content.pm.ServiceInfo.FOREGROUND_SERVICE_TYPE_MICROPHONE;
^
  symbol:   static FOREGROUND_SERVICE_TYPE_MICROPHONE
  location: class ServiceInfo

/Users/ad/MY COOL APP/node_modules/react-native-callkeep/android/src/main/java/io/wazo/callkeep/VoiceConnectionService.java:251: error: cannot find symbol
        startForeground(FOREGROUND_SERVICE_TYPE_MICROPHONE, notification);
                        ^
  symbol:   variable FOREGROUND_SERVICE_TYPE_MICROPHONE
  location: class VoiceConnectionService
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors
```

- Reproduced on:
 - [x] Android
 - [ ] iOS

## Steps to Reproduce
Install the package, follow the installation process https://github.com/react-native-webrtc/react-native-callkeep/blob/master/docs/android-installation.md

## Versions
    - Callkeep: "^4.0.0"
    - React Native: "0.63.2"

**Comment by Pauligrinder** [comment]
I solved this by going to `/Users/ad/MY COOL APP/node_modules/react-native-callkeep/android/src/main/java/io/wazo/callkeep/VoiceConnectionService.java` and commenting out `import static android.content.pm.ServiceInfo.FOREGROUND_SERVICE_TYPE_MICROPHONE;` , and then replacing `FOREGROUND_SERVICE_TYPE_MICROPHONE` with `128` (which is what the static value gets from the Android system).

**Comment by manuquentin** [comment]
Thanks @ajitdas123 for the report, which version of Android do you have ? With which version of the android sdk do you compile ?
We should maybe replace `FOREGROUND_SERVICE_TYPE_MICROPHONE` by `128` as suggested by @Pauligrinder for sdk <= 29.

**Comment by Pauligrinder** [comment]
I'm using these versions, if that helps:

`buildToolsVersion = "29.0.2"
        minSdkVersion = 16
        compileSdkVersion = 29
        targetSdkVersion = 29`

**Comment by devWaleed** [comment]
Same issue here on Android, need a hot PR.

**Comment by stephanoparaskeva** [comment]
Same issue here

**Comment by MrHertal** [comment]
Same issue with Android SDK 29
I rolled back to 3.1.1

**Comment by jason-shen** [comment]
any update on this? i have the same issue

**Comment by manuquentin** [comment]
I'll publish a fix today

**Comment by jason-shen** [comment]
> I'll publish a fix today

tested all works, also the setting of build.gradle has to be that as well

buildToolsVersion = '30'
        minSdkVersion = 27
        compileSdkVersion = 30
        targetSdkVersion = 30

or else the forgroudServiceType won't work on anything lower than 30.

i am trying on a note10 5g, its on android 10, but this was needed to get access to the mic when the app is killed and locked

cheers for the fix @manuquentin

**Comment by manuquentin** [comment]
Fix in `4.0.1`

**Comment by MrHertal** [comment]
thanks @manuquentin !
